### PR TITLE
Unofficial grimoire fix 

### PR DESCRIPTION
**Pull request by silentmark**
There was an issue with overcast not working due to data not being pasted from dialog to test and also calculation of user entries: 

```
        for(let key in this.userEntry)
        {
            if (["string", "boolean"].includes(typeof this.userEntry[key]))
            {
                this.fields[key] = this.userEntry[key]
            }
            else if (Number.isNumeric(this.userEntry[key]))
            {
                this.fields[key] += this.userEntry[key];
            }
        }
        this.tooltips.finish(this, "User Entry")
```

since overcast value in cast dialog is a numeric field, instead of passing it to test, it was either incremented on every input change, or treated as NaN, due to null + number calculation.

Also, minor typo and translation in base dialog.

### CODE REVIEW

**Comment by silentmark** [comment]
@moo-man - this PR has been updated to latest changes.

File: modules/apps/roll-dialog/cast-dialog.js [hunk 1]
```diff
@@ -13,6 +13,13 @@ export default class CastDialog extends SkillDialog {
         return options;
     }
 
+    constructor(fields, data, resolve, options) {
+        super(fields, data, resolve, options);
+
+        this.initialFields.overchannelling = this.fields.overchannelling ?? 0;
+        this.fields.overchannelling = this.fields.overchannelling ?? 0;
+    }
```
Comment (moo-man) [inline modules/apps/roll-dialog/cast-dialog.js:21]: Is there a reason this isn't part of an overridden `_defaultFields` function? See https://github.com/moo-man/WFRP4e-FoundryVTT/blob/master/modules/apps/roll-dialog/roll-dialog.js#L184

File: modules/apps/roll-dialog/cast-dialog.js [hunk 2]
```diff
@@ -69,6 +76,18 @@ export default class CastDialog extends SkillDialog {
...
 
+    async getData() 
+    {
+        let context = await super.getData();
+        if (this.userEntry.overchannelling) {
+            let overchannelling = parseInt(this.userEntry.overchannelling) ?? 0;
+            this.userEntry.overchannelling = overchannelling;
+            this.fields.overchannelling = overchannelling;
+            context.fields.overchannelling = overchannelling;
+        }
```
Comment (moo-man) [inline modules/apps/roll-dialog/cast-dialog.js:87]: Not sure what the point of this is? This should all be handled natively as any other field.

File: static/templates/dialog/base-dialog.hbs [hunk 3]
```diff
@@ -27,7 +27,7 @@
       <label>{{localize "DIALOG.DialogModifiers"}}</label>
       <ul class="modifier-list">
         {{#each data.scripts}}
-            <li class="modifier {{#if this.isActive}}active {{else if this.isHidden}}hidden{{/if}}" data-tooltip="{{this.effect.name}} {{#if this.options.targeter}}(Target){{/if}}" data-tooltip-direction="RIGHT" data-index="{{@key}}" >&#8226; <a>{{this.Label}}</a></li>
+            <li class="modifier {{#if this.isActive}}active {{else if this.isHidden}}hidden{{/if}}" data-tooltip="{{this.effect.name}} {{#if this.options.targeter}}({{localize 'Target'}}){{/if}}" data-tooltip-direction="RIGHT" data-index="{{@key}}" >&#8226; <a>{{this.Label}}</a></li>
         {{/each}}
       </ul>
     </div>
```
Comment (moo-man) [inline static/templates/dialog/base-dialog.hbs:33]: This should probably just use the partial provided by the warhammer library (which needs to be localized too)

File: static/templates/dialog/spell-dialog.hbs [hunk 4]
```diff
@@ -4,25 +4,27 @@
...
-    </label>
+    <input class="input-text" type="text" name="overchannelling" value="{{fields.overchannelling}}" data-dtype="Number" title="{{localize 'DIALOG.OverchannellingTooltip'}}" />
   </div>
+  {{#if false}}
+    <div class="form-group">
+      <label title="{{localize 'DIALOG.Quickcasting'}}">{{localize "DIALOG.Quickcasting"}}
+        <input title="{{localize 'DIALOG.QuickcastingTooltip'}}" type="checkbox" name="quickcasting" data-dtype="Boolean" {{checked fields.quickcasting}} />
+      </label>
+    </div>
+  {{/if}}
```
Comment (moo-man) [inline static/templates/dialog/spell-dialog.hbs:29]: I don't really use `title` or `data-dtype` anymore.

File: modules/apps/roll-dialog/cast-dialog.js (hunk 1 above)
Comment (silentmark) [reply modules/apps/roll-dialog/cast-dialog.js:21]: I will test it out

File: modules/apps/roll-dialog/cast-dialog.js (hunk 2 above)
Comment (silentmark) [reply modules/apps/roll-dialog/cast-dialog.js:87]: There was a problem with Numeric fields for userEntry. Maybe it's now handled better in Library. The problem was, that overchannelling was always treated as string, and in result, onFieldChange was returning either NaN as text or undefined. I will test it out.

File: static/templates/dialog/base-dialog.hbs (hunk 3 above)
Comment (silentmark) [reply static/templates/dialog/base-dialog.hbs:33]: it seams that none of the partial templates are being used in WFRP4e system so far. as of Lib translation, I might need to prepare PR there anyway.

File: static/templates/dialog/spell-dialog.hbs (hunk 4 above)
Comment (silentmark) [reply static/templates/dialog/spell-dialog.hbs:29]: copy-paste, i will remove those.

File: modules/apps/roll-dialog/cast-dialog.js (hunk 1 above)
Comment (silentmark) [reply modules/apps/roll-dialog/cast-dialog.js]: resolved

File: modules/apps/roll-dialog/cast-dialog.js (hunk 2 above)
Comment (silentmark) [reply modules/apps/roll-dialog/cast-dialog.js]: resolved, it works fine without this change.

**Comment by silentmark** [comment]
@moo-man - went through comments and updated PR.

**Pull request merged**